docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/07_ReactJS_Redux_and_GraphQL.md

```diff
@@ -0,0 +1,666 @@
+  switch (action.type) {
+  case 'INCREMENT':
+    return state + 1
+  case 'DECREMENT':
+    return state - 1
+  default:
+    return state
+  }
+}
+
+let store = createStore(counter)
+store.subscribe(() =>
+  console.log(store.getState())
+)
+// Call an action
+store.dispatch({ type: 'INCREMENT' })
+// 1
+```
+
+## Redux Devtools
+
+It's important to be able to view the state of the React application when you're debugging and
+building the interface.
+
+To be able to view the state, you'll need to be in a dev environment 
+and have the [Redux Devtools](https://github.com/zalmoxisus/redux-devtools-extension)
+installed on Google Chrome or Firefox, which can be found by searching with your favourite search
+engine.
+
+
+## GraphQL and Apollo
+
+[GraphQL](http://graphql.org/learn/) is a strictly-types query language that allows you to describe what data you want to fetch from your API. Because it is based on types, it is self-documenting and predictable. Further, it's structure lends itself nicely to fetching nested objects. Here is an example of a simple GraphQL query:
+
+```
+query GetUser($ID: Int!) {
+	user {
+		name
+		email
+		blogPosts {
+			title
+			comments(Limit: 5) {
+				author
+				comment
+			}
+		}
+
+	}
+}
+```
+
+The above query is almost self-descriptive. It gets a user by ID, returns his or her name and email address, along with the title of any blog posts he or she has written, and the first five comments for each of those. The result of that query is, very predictably, JSON that takes on the same structure.
+
+```js
+{
+	"user": {
+		"name": "Test user",
+		"email": "me@example.com",
+		"blogPosts": [
+			{
+				"title": "How to be awesome at GraphQL",
+				"comments": [
+					{
+						"author": "Uncle Cheese",
+						"comment": "Nice stuff, bro"
+					}
+				]
+			}
+		]
+	}
+}
+```
+
+On its own, GraphQL offers nothing functional, as it's just a query language. You still need a service that will invoke queries and map their results to UI. For that, SilverStripe uses an implementation of [Apollo](http://dev.apollodata.com/) that works with React.
+
+## For more information
+
+This documentation will stop short of explaining React, Redux, and GraphQL/Apollo in-depth, as there is much better
+documentation available all over the web. We recommend:
+* [The Official React Tutorial](https://facebook.github.io/react/tutorial/tutorial.html)
+* [Build With React](http://buildwithreact.com/tutorial)
+* [Getting Started with Redux](https://egghead.io/courses/getting-started-with-redux)
+* [The React Apollo docs](http://dev.apollodata.com/react/)
+
+# The Injector API
+
+Much like customisation is done in the backend through [Injector](../../extending/injector), the client-side framework has its own implementation of dependency injection known as `Injector`. Using Injector, you can register new services, and transform existing services.
+
+Injector is broken up into three sub-APIs:
+* `Injector.react` for React UI components
```

Comment:
It's a bit weird that we're naming one API after a library (`react`), and the other one after a concept (`reducer`). Why not `component` and `reducer`? You've probably discussed this already, just wondering.